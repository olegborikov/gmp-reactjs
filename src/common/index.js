export const navigateToSearch = (history, searchQuery, searchParams) =>
  history.push(`/search${searchQuery ? "/" + searchQuery : ""}?${searchParams.toString()}`)

