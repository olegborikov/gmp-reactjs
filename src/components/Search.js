import {Col, Input} from 'antd';
import {useCallback} from "react";

function Search({onNameSearch}) {

  const handleInputChange = useCallback(event => {
    onNameSearch(event.target.value)
  }, [onNameSearch])

  return (
    <Col span={10} offset={10}>
      <Input onChange={handleInputChange}/>
    </Col>
  );
}

export default Search;