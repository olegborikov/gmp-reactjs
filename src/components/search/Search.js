import React, {useState} from "react";
import SearchInput from "./input/SearchInput";
import Button from "../common/button/Button";
import classes from "./Search.module.css";
import {useNavigate, useParams, useSearchParams} from "react-router-dom";
import {navigateToSearch} from "../../common";

function Search() {
  const navigate = useNavigate();
  const [searchParams] = useSearchParams();
  const {searchQuery} = useParams();
  const [search, setSearch] = useState(searchQuery ? searchQuery : "")
  return (
    <div className={classes.search}>
      <SearchInput search={search} setSearch={setSearch}/>
      <Button name="SEARCH" size="large" action={() => navigateToSearch(navigate, search, searchParams)}/>
    </div>
  );
}

export default Search
