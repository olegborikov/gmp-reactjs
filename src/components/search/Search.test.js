import React from "react";
import {fireEvent, render, screen} from "@testing-library/react";
import Search from "./Search";
import {navigateToSearch} from "../../common";

jest.mock("react-router-dom", () => {
  return {
    __esModule: true,
    useNavigate: jest.fn(),
    useSearchParams: jest.fn(() => [{}]),
    useParams: jest.fn(() => ({
      searchQuery: ""
    }))
  }
})

jest.mock("../../common");

it("search test", () => {
  const {asFragment} = render(<Search/>)
  expect(asFragment()).toMatchSnapshot()

  fireEvent.change(screen.getByRole("textbox"), {
    target: {value: "Movie name"}
  })
  expect(screen.getByDisplayValue("Movie name")).toBeInTheDocument

  fireEvent.click(screen.getByRole("button"))

  expect(navigateToSearch).toHaveBeenCalled()
})
