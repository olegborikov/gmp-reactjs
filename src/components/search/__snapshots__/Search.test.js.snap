// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search test 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <input
      class="input"
      value=""
    />
    <div
      class="confirm"
    >
      <button
        class="large 
              red"
        type=""
      >
        SEARCH
      </button>
    </div>
  </div>
</DocumentFragment>
`;
